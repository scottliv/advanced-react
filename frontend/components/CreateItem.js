import React, { Component } from "react";
import { Mutation } from "react-apollo";
import gql from "graphql-tag";
import Router from "next/router";
import Form from "./styles/Form";
import Error from "./ErrorMessage";
import formatMoney from "../lib/formatMoney";

const CREATE_ITEM_MUTATION = gql`
  mutation CREATE_ITEM_MUTATION($data: createItemInput) {
    createItem(data: $data) {
      id
      title
      description
      image
      largeImage
      price
    }
  }
`;

class CreateItem extends Component {
  state = {
    createItemInput: {
      title: "",
      description: "",
      price: 0,
      image: "",
      largeImage: ""
    },
    imageLoading: false
  };
  handleChange = e => {
    const { name, type, value } = e.target;
    let val = type === "number" ? parseFloat(value) : value;
    this.setState({
      ...this.state,
      createItemInput: { ...this.state.createItemInput, [name]: val }
    });
  };

  uploadImage = async e => {
    this.setState({ ...this.state, imageLoading: true });
    e.preventDefault();
    const cloudName = "dxev349q4";
    const url = `https://api.cloudinary.com/v1_1/${cloudName}/image/upload`;
    const files = e.target.files;
    const formData = new FormData();

    if (files.length) {
      formData.append("file", files[0]);
      formData.append("upload_preset", "sickfits");
      try {
        const res = await fetch(url, {
          method: "POST",
          body: formData
        });
        const file = await res.json();
        this.setState({
          ...this.state,
          createItemInput: {
            ...this.state.createItemInput,
            image: file.secure_url,
            largeImage: file.eager[0].secure_url
          },
          imageLoading: false
        });
      } catch {
        throw new Error("There was a problem updloading your image");
      }
    } else {
      this.setState({ createItemInput: { image: "" } });
    }
  };
  render() {
    return (
      <Mutation
        mutation={CREATE_ITEM_MUTATION}
        variables={{ data: { ...this.state.createItemInput } }}
      >
        {(createItem, { error, loading }) => {
          return (
            <Form
              onSubmit={async e => {
                e.preventDefault();
                const res = await createItem();
                console.log(res);
                Router.push({
                  pathname: "/item",
                  query: { id: res.data.createItem.id }
                });
              }}
            >
              <h2>Sell An Item</h2>
              <Error error={error} />
              <fieldset
                disabled={loading && this.state.imageLoading}
                aria-busy={loading && this.state.imageLoading}
              >
                <label htmlFor="file">
                  Image
                  <input
                    type="file"
                    id="file"
                    name="file"
                    placeholder="Upload an image"
                    onChange={this.uploadImage}
                  />
                  {this.state.createItemInput.image && (
                    <img
                      width="200"
                      src={this.state.createItemInput.image}
                      alt="Item Image"
                    />
                  )}
                </label>
                <label htmlFor="title">
                  Title
                  <input
                    type="text"
                    id="title"
                    name="title"
                    placeholder="Title"
                    required
                    value={this.state.createItemInput.title}
                    onChange={this.handleChange}
                  />
                </label>
                <label htmlFor="price">
                  Price
                  <input
                    type="number"
                    id="price"
                    name="price"
                    placeholder="price"
                    required
                    value={this.state.createItemInput.price}
                    onChange={this.handleChange}
                  />
                </label>
                <label htmlFor="description">
                  Description
                  <textarea
                    type="text"
                    id="description"
                    name="description"
                    placeholder="description"
                    required
                    value={this.state.createItemInput.description}
                    onChange={this.handleChange}
                  />
                </label>
                <button type="submit">Submit</button>
              </fieldset>
            </Form>
          );
        }}
      </Mutation>
    );
  }
}

export default CreateItem;
export { CREATE_ITEM_MUTATION };
