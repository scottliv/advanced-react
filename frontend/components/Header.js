import React from "react";
import Router from "next/router";
import NProgress from "nprogress";
import Meta from "./Meta";
import Nav from "./Nav";

import Link from "next/link";
import styled from "styled-components";

Router.onRouteChangeStart = () => {
  console.log("triggered");
  NProgress.start();
};
Router.onRouteChangeComplete = () => {
  NProgress.done();
  console.log("triggered");
};
Router.onRouteChangeError = () => {
  NProgress.done();
  console.log("triggered");
};
const Logo = styled.h1`
  font-size: 4rem;
  margin-left: 2rem;
  position: relative;
  z-index: 2;
  transform: skew(-7deg);
  a {
    padding: 0.5rem 1rem;
    background: ${props => props.theme.red};
    text-transform: uppercase;
    text-decoration: none;
    color: #fff;
  }
  @media (max-width: ${props => props.theme.maxWidth}) {
    margin: 0;
    text-align: center;
  }
`;

const StyledHeader = styled.header`
  .bar {
    border-bottom: 10px solid ${props => props.theme.black};
    display: grid;
    grid-template-columns: auto 1fr;
    justify-content: space-between;
    align-items: stretch;
    @media (max-width: ${props => props.theme.maxWidth}) {
      grid-template-columns: 1fr;
      justify-content: center;
    }
  }
  .sub-bar {
    grid-template-columns: 1fr auto;
    border-bottom: 1px solid ${props => props.theme.lightGrey};
  }
`;

const Header = () => {
  return (
    <StyledHeader className="header">
      <Meta />
      <div className="bar">
        <Logo>
          <Link href="/">
            <a>Sick Fits</a>
          </Link>
        </Logo>
        <Nav />
      </div>
      <div className="sub-bar">
        <p>Search</p>
        <p>Cart</p>
      </div>
    </StyledHeader>
  );
};

export default Header;
