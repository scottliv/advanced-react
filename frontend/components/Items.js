import React, { Component } from "react";
import { Query } from "react-apollo";
import gql from "graphql-tag";
import styled from "styled-components";
import Pagination from "./Pagination";
import { perPage } from "../config";
import Item from "./Item";

const Centered = styled.div`
  text-align: center;
  margin: 0 auto;
`;

const ItemsList = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 60px;
  max-width: ${props => props.theme.maxWidth};
`;
const ALL_ITEMS_QUERY = gql`
  query ALL_ITEMS_QUERY($skip: Int = 0, $first: Int = ${perPage} ) {
    items(skip: $skip, first: $first, orderBy: createdAt_DESC) {
      id
      price
      title
      description
      image
      largeImage
    }
  }
`;

class Items extends Component {
  render() {
    return (
      <Centered>
        <Pagination page={this.props.page} />
        <Query
          query={ALL_ITEMS_QUERY}
          variables={{
            skip: this.props.page * perPage - perPage,
            first: perPage
          }}
        >
          {({ loading, error, data }) => {
            if (loading) {
              return <p>Loading</p>;
            }
            if (error) {
              return <p>Error: {error.message}</p>;
            }

            return (
              <ItemsList>
                {data.items.map(item => {
                  return <Item item={item} key={item.id} />;
                })}
              </ItemsList>
            );
          }}
        </Query>
        <Pagination page={this.props.page} />
      </Centered>
    );
  }
}

export default Items;
export { ALL_ITEMS_QUERY };
