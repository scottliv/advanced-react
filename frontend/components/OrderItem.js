import React, { Component } from "react";
import PropTypes from "prop-types";
import formatMoney from "../lib/formatMoney";

class OrderItem extends Component {
  render() {
    const { item } = this.props;
    return (
      <div className="order-item">
        {item.image && <img src={item.image} alt={item.title} />}

        <div className="item-details">
          <h2>{item.title}</h2>
          <p>QTY: {item.quantity}</p>
          <p>Each: {formatMoney(item.price)}</p>
          <p>Subtotal: {formatMoney(item.price)}</p>
          <p>{item.description}</p>
        </div>
      </div>
    );
  }
}

OrderItem.propTypes = {
  item: PropTypes.object.isRequired
};

export default OrderItem;
