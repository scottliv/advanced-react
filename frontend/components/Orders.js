import React, { Component } from "react";
import { Query } from "react-apollo";
import Link from "next/link";
import styled from "styled-components";
import { formatDistance } from "date-fns";
import gql from "graphql-tag";
import Error from "./ErrorMessage";
import OrderItemStyles from "./styles/OrderItemStyles";
import formatMoney from "../lib/formatMoney";

const ORDERS_QUERY = gql`
  query ORDERS_QUERY {
    orders(orderBy: createdAt_DESC) {
      id
      total
      createdAt
      items {
        id
        title
        description
        price
        quantity
        image
      }
    }
  }
`;

const OrdersUl = styled.ul`
  display: grid;
  grid-gap: 4rem;
  grid-template-columns: repeat(auto-fit, minmax(40%, 1fr));
`;

class Orders extends Component {
  render() {
    return (
      <Query query={ORDERS_QUERY}>
        {({ data: { orders }, error, loading }) => {
          if (error) return <Error error={error} />;
          if (loading) return <p>Loading...</p>;
          return (
            <div>
              <h2>You have {orders.length} orders.</h2>
              <OrdersUl>
                {orders.map(order => (
                  <OrderItemStyles key={order.id}>
                    <Link
                      href={{
                        pathname: "/order",
                        query: { id: order.id }
                      }}
                    >
                      <a>
                        <div className="order-meta">
                          <p>
                            {order.items.reduce((a, b) => a + b.quantity, 0)}{" "}
                            Items
                          </p>
                          <p>{order.items.lenth} Products</p>
                          <p>{formatDistance(order.createdAt, new Date())}</p>
                          <p>{formatMoney(order.total)}</p>
                        </div>
                        <div className="images">
                          {order.items.map(item => (
                            <img
                              src={item.image}
                              alt={item.title}
                              key={item.id}
                            />
                          ))}
                        </div>
                      </a>
                    </Link>
                  </OrderItemStyles>
                ))}
              </OrdersUl>
            </div>
          );
        }}
      </Query>
    );
  }
}

export default Orders;
