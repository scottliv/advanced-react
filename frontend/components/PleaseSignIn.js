import { Query } from "react-apollo";
import { CURRENT_USER_QUERY } from "./User";
import SignIn from "./SignIn";

const PleaseSignIn = props => (
  <Query query={CURRENT_USER_QUERY}>
    {({ data, loading }) => {
      if (loading) return <p>Loading...</p>;
      if (!data.me) {
        return (
          <div>
            <p>You must be logged in to view this page</p>
            <SignIn />
          </div>
        );
      }
      return props.children;
    }}
  </Query>
);

export default PleaseSignIn;
