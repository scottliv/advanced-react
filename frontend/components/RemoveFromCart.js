import React from "react";
import { Mutation } from "react-apollo";
import gql from "graphql-tag";
import styled from "styled-components";
import PropTypes from "prop-types";
import { CURRENT_USER_QUERY } from "./User";

const BigButton = styled.button`
  font-size: 3rem;
  background: none;
  border: none;
  &:hover {
    color: ${props => props.theme.red};
    cursor: pointer;
  }
`;
const REMOVE_FROM_CART_MUTATION = gql`
  mutation removeFromCart($id: ID!) {
    removeFromCart(id: $id) {
      id
    }
  }
`;
const update = (cache, payload) => {
  const data = cache.readQuery({ query: CURRENT_USER_QUERY });
  const cartItemId = payload.data.removeFromCart.id;
  data.me.cart = data.me.cart.filter(cartItem => cartItem.id !== cartItemId);
  cache.writeQuery({ query: CURRENT_USER_QUERY, data });
};

const RemoveFromCart = id => (
  <Mutation
    mutation={REMOVE_FROM_CART_MUTATION}
    variables={id}
    update={update}
    optimisticResponse={{
      __typename: "Mutation",
      removeFromCart: {
        __typename: "CartItem",
        id: id
      }
    }}
    // refetchQueries={[{ query: CURRENT_USER_QUERY }]}
  >
    {(removeFromCart, { loading }) => (
      <BigButton
        disabled={loading}
        onClick={removeFromCart}
        title="Delete Item"
      >
        &times;
      </BigButton>
    )}
  </Mutation>
);

RemoveFromCart.propTypes = {
  id: PropTypes.string.isRequired
};

export default RemoveFromCart;
