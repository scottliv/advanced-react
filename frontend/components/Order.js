import React, { Component } from "react";
import PropTypes from "prop-types";
import { Query } from "react-apollo";
import { format } from "date-fns";
import Head from "next/head";
import gql from "graphql-tag";
import formatMoney from "../lib/formatMoney";
import Error from "./ErrorMessage";
import OrderStyles from "./styles/OrderStyles";
import OrderItem from "./OrderItem";

const SINGLE_ORDER_QUERY = gql`
  query SINGLE_ORDER_QUERY($id: ID!) {
    order(id: $id) {
      id
      charge
      total
      createdAt
      user {
        id
      }
      items {
        id
        title
        description
        price
        quantity
        image
      }
    }
  }
`;

class Order extends Component {
  static propTypes = {
    id: PropTypes.string.isRequired
  };
  render() {
    return (
      <Query query={SINGLE_ORDER_QUERY} variables={{ id: this.props.id }}>
        {(data, error, loading) => {
          if (error) return <Error error={error} />;
          if (loading) return <p>Loading...</p>;
          const order = data.data.order;
          console.log(order);
          return (
            <OrderStyles>
              <Head>
                <title>Sick Fits Order: {order && order.id}</title>
              </Head>
              <p>
                <span>Order:</span>
                <span>{this.props.id}</span>
              </p>
              <p>
                <span>Charge</span>
                <span>{order.charge}</span>
              </p>

              <p>
                <span>Date</span>
                <span>
                  {order && format(order.createdAt, "MMMM d, YYYY h:mm a")}
                </span>
              </p>
              <p>
                <span>Total</span>
                <span>{order && formatMoney(order.total)}</span>
              </p>
              <p>
                <span>Item Count:</span>
                <span>{order.items.length}</span>
              </p>
              <div className="items">
                {order.items.map(item => (
                  <OrderItem item={item} key={item.id} />
                ))}
              </div>
            </OrderStyles>
          );
        }}
      </Query>
    );
  }
}

export default Order;
