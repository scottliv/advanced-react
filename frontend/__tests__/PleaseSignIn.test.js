import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import { MockedProvider } from "react-apollo/test-utils";
import wait from "waait";
import PleaseSignIn from "../components/PleaseSignIn";
import { CURRENT_USER_QUERY } from "../components/User";
import { fakeUser } from "../lib/testUtils";

const notSignedInMocks = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: { data: { me: null } }
  }
];

const signedInMocks = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: { data: { me: fakeUser() } }
  }
];

describe("<PleaseSignIn/>", () => {
  it("renders login component when user not signed in", async () => {
    const wrapper = mount(
      <MockedProvider mocks={notSignedInMocks}>
        <PleaseSignIn />
      </MockedProvider>
    );
    expect(wrapper.text()).toContain("Loading...");
    await wait();
    wrapper.update();
    expect(toJson(wrapper.find("p"))).toMatchSnapshot();
    expect(toJson(wrapper.find("SignIn"))).toMatchSnapshot();
  });

  it("renders the child component when the user is signed in", async () => {
    const Hey = () => <p>Hey!</p>;
    const user = fakeUser();

    const wrapper = mount(
      <MockedProvider mocks={signedInMocks}>
        <PleaseSignIn>
          <Hey />
        </PleaseSignIn>
      </MockedProvider>
    );

    await wait();
    wrapper.update();
    expect(wrapper.contains(<Hey />)).toBe(true);
  });
});
