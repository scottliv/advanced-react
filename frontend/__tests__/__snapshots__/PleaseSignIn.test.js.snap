// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PleaseSignIn/> renders login component when user not signed in 1`] = `
<p>
  Please Sign In before Continuing
</p>
`;

exports[`<PleaseSignIn/> renders login component when user not signed in 2`] = `
<SignIn>
  <Mutation
    mutation={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "SIGNIN_MUTATION",
            },
            "operation": "mutation",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "email",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "email",
                        },
                      },
                    },
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "password",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "password",
                        },
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "signin",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "email",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "name",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [
              Object {
                "defaultValue": undefined,
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "email",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "password",
                  },
                },
              },
            ],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 158,
          "start": 0,
        },
      }
    }
    refetchQueries={
      Array [
        Object {
          "query": Object {
            "definitions": Array [
              Object {
                "directives": Array [],
                "kind": "OperationDefinition",
                "name": undefined,
                "operation": "query",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "me",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "email",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "name",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "permissions",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "cart",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                  "selectionSet": undefined,
                                },
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "quantity",
                                  },
                                  "selectionSet": undefined,
                                },
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "item",
                                  },
                                  "selectionSet": Object {
                                    "kind": "SelectionSet",
                                    "selections": Array [
                                      Object {
                                        "alias": undefined,
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "id",
                                        },
                                        "selectionSet": undefined,
                                      },
                                      Object {
                                        "alias": undefined,
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "price",
                                        },
                                        "selectionSet": undefined,
                                      },
                                      Object {
                                        "alias": undefined,
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "image",
                                        },
                                        "selectionSet": undefined,
                                      },
                                      Object {
                                        "alias": undefined,
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "title",
                                        },
                                        "selectionSet": undefined,
                                      },
                                      Object {
                                        "alias": undefined,
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "description",
                                        },
                                        "selectionSet": undefined,
                                      },
                                    ],
                                  },
                                },
                              ],
                            },
                          },
                        ],
                      },
                    },
                  ],
                },
                "variableDefinitions": Array [],
              },
            ],
            "kind": "Document",
            "loc": Object {
              "end": 238,
              "start": 0,
            },
          },
        },
      ]
    }
    variables={
      Object {
        "email": "",
        "password": "",
      }
    }
  >
    <Form
      method="post"
      onSubmit={[Function]}
    >
      <form
        className="Form-sc-1xszr8q-0 krRxky"
        method="post"
        onSubmit={[Function]}
      >
        <fieldset
          aria-busy={false}
          disabled={false}
        >
          <h2>
            Sign In to your account
          </h2>
          <DisplayError
            error={Object {}}
          />
          <label
            htmlFor="email"
          >
            Email
            <input
              name="email"
              onChange={[Function]}
              placeholder="email"
              type="email"
              value=""
            />
          </label>
          <label
            htmlFor="password"
          >
            Password
            <input
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
            />
          </label>
        </fieldset>
        <button
          type="submit"
        >
          Submit
        </button>
      </form>
    </Form>
  </Mutation>
</SignIn>
`;
